stl/inc/xutility

```diff
@@ -1001,15 +1023,16 @@ namespace ranges {
         private:
             enum class _St { _None, _Custom, _Swap, _Exchange };
 
-            // clang-format off
             template <class _Ty1, class _Ty2>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
                 if constexpr (_Has_ADL<_Ty1, _Ty2>) {
                     return {_St::_Custom, noexcept(iter_swap(_STD declval<_Ty1>(), _STD declval<_Ty2>()))};
                 } else if constexpr (_Can_swap_references<_Ty1, _Ty2>) {
                     return {_St::_Swap, noexcept(_RANGES swap(*_STD declval<_Ty1>(), *_STD declval<_Ty2>()))};
-                } else if constexpr (indirectly_movable_storable<remove_reference_t<_Ty1>, remove_reference_t<_Ty2>>
-                    && indirectly_movable_storable<remove_reference_t<_Ty2>, remove_reference_t<_Ty1>>) {
+                } else if constexpr (

```

Comment:
I guess the previous version was better. Maybe add a comment?

Reply:
I really hate what clang-format does to boolean expressions - I suspect it's intentionally horrible to encourage people not to write complex boolean expressions.

Reply:
There is already a PR for that but i guess we need another year until it is reviewed

Reply:
I pulled this nasty expression out into a helper concept, which both makes this readable and can improve throughput in cases where the first term is `false`.